governance/src/agent.js

```diff
@@ -0,0 +1,301 @@
+      contractAddress,
+      eventName,
+    },
+  });
+}
+
+// alert for when a new governance proposal is created
+function proposalCreatedFinding(proposal, address, config) {
+  return Finding.fromObject({
+    name: `${config.protocolName} Governance Proposal Created`,
+    description: `Governance Proposal ${proposal.id} was just created`,
+    alertId: `${config.developerAbbreviation}-${config.protocolAbbreviation}-PROPOSAL-CREATED`,
+    type: 'Info',
+    severity: 'Info',
+    protocol: config.protocolName,
+    metadata: {
+      address,
+      ...proposal,
+    },
+  });
+}
+
+function voteEmittedFinding(voteInfo, address, config) {
+  console.log(voteInfo);
+  let description = `Vote emitted with weight ${voteInfo.votingPower.toString()}`;
+  if (voteInfo.support) {
+      description += ' in support of';
+  } else {
+      description += ' against';
+  }
+  description += ` proposal ${voteInfo.id}`;
+
+  return Finding.fromObject({
+    name: `${config.protocolName} Governance Proposal Vote Emitted`,
+    description,
+    alertId: `${config.developerAbbreviation}-${config.protocolAbbreviation}-VOTE-EMITTED`,
+    type: 'Info',
+    severity: 'Info',
+    protocol: config.protocolName,
+    metadata: {
+      id: voteInfo.id.toString(),
+      address,
+      voter: voteInfo.voter,
+      weight: voteInfo.votingPower.toString(),
+    },
+  });
+}
+
+function proposalCanceledFinding(proposalId, address, config) {
+  return Finding.fromObject({
+    name: `${config.protocolName} Governance Proposal Canceled`,
+    description: `Governance proposal ${proposalId} has been canceled`,
+    alertId: `${config.developerAbbreviation}-${config.protocolAbbreviation}-PROPOSAL-CANCELED`,
+    type: 'Info',
+    severity: 'Info',
+    protocol: config.protocolName,
+    metadata: {
+      address,
+      id: proposalId,
+      state: 'canceled',
+    },
+  });
+}
+
+function proposalExecutedFinding(proposalId, address, config) {
+  return Finding.fromObject({
+    name: `${config.protocolName} Governance Proposal Executed`,
+    description: `Governance proposal ${proposalId} has been executed`,
+    alertId: `${config.developerAbbreviation}-${config.protocolAbbreviation}-PROPOSAL-EXECUTED`,
+    type: 'Info',
+    severity: 'Info',
+    protocol: config.protocolName,
+    metadata: {
+      address,
+      id: proposalId,
+      state: 'executed',
+    },
+  });
+}
+
+function proposalQueuedFinding(proposalId, address, config, eta) {
+  return Finding.fromObject({
+    name: `${config.protocolName} Governance Proposal Queued`,
+    description: `Governance proposal ${proposalId} has been queued`,
+    alertId: `${config.developerAbbreviation}-${config.protocolAbbreviation}-PROPOSAL-QUEUED`,
+    type: 'Info',
+    severity: 'Info',
+    protocol: config.protocolName,
+    metadata: {
+      address,
+      id: proposalId,
+      state: 'queued',
+    },
+  });
+}
+
+function quorumNumeratorUpdatedFinding(address, config, oldNum, newNum) {
```

Comment:
this function was never used - you could add a `case` statement on line `261-284` where you are filtering for log events and create a new case/alert for this